src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -49,38 +73,91 @@ public abstract class AssemblyLoadContext
         internal static extern void InternalStartProfile(string profile, IntPtr ptrNativeAssemblyLoadContext);
 #endif // FEATURE_MULTICOREJIT
 
-        protected AssemblyLoadContext()
+        static AssemblyLoadContext()
         {
-            // Initialize the ALC representing non-TPA LoadContext
-            InitializeLoadContext(false);
+            // We register the cleanup of all AssemblyLoadContext that have not been finalized in the AppContext.Unloading
+            AppContext.Unloading += OnAppContextUnloading;
         }
 
-        internal AssemblyLoadContext(bool fRepresentsTPALoadContext)
+#if FEATURE_COLLECTIBLE_ALC
+        protected AssemblyLoadContext() : this(false, false)
         {
-            // Initialize the ALC representing TPA LoadContext
-            InitializeLoadContext(fRepresentsTPALoadContext);
         }
-        
-        void InitializeLoadContext(bool fRepresentsTPALoadContext)
+
+        protected AssemblyLoadContext(bool isCollectible) : this(false, isCollectible)
```

Comment:
I thought that coreclr and corefx would be built with FEATURE_COLLECTIBLE_ALC activated or deactivated for both at the same time? (since I have added FEATURE_COLLECTIBLE_ALC to also the contract on corefx PR)